doc/developer-guides/modularity.rst

```diff
@@ -0,0 +1,89 @@
+.. _modularity:
+
+ACRN Hypervisor: Modular Design
+###############################
+
+Overview
+********
+
+ACRN highly emphasizes modular design, i.e. the separation of functionality info
+modules that define a concise set of interfaces. The goals of modular design
+include:
+
+* **Understandability** A modular design is easier to understand due to
+  encapsulation.
+* **Testability** Modules can be integrated and tested in the reverse order of
+  dependencies among them. White-box integration tests help improve the coverage
+  of tests and identify corner cases that are hard to trigger when testing the
+  hypervisor as a whole.
+* **Configurability** Modular design makes it easy to configure certain
+  functionalities in or out. This is crucial in safety-critical scenarios
+  because absence of irrelevant code is required in both MISRA-C and functional
+  safety standards.
+* **Meet functional safety requirements** Functional safety standards explicitly
+  requires a hierarchical structure of modules in software architectural
+  design. This applies to any safety integrity level defined in IEC 61508
```

Comment:
Spaces aren't allowed in citation references, so change to:
```
... defined in [IEC_61508-3]_ or [ISO_ 26262-6]_.
```

Reply:
Updated. Thanks.